### PR TITLE
2125 fix null reference exception

### LINKED ISSUE #2125: NullReferenceException in SymbolTable.GetParagraph / CodeElementFinder
**Describe the bug**

Following code produces 2 exception:
image
image

**To Reproduce**
(Type)Cobol code that cause the bug : (if any)
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. MyPGM.
       PROCEDURE DIVISION
                          .
       PARA0.
           IF  OK
              SEARCH ALL MyArray 
      
              END-SEARCH
           END-IF
      
           PERFORM PARA1
      
           .
       PARA1.
      
           .
      
       END PROGRAM ABIT0013.
```

Expected (Type)Cobol code : 
No error.

### PR DESCRIPTION
**Pull request by laabidiHend**
Fixes #2125

### CODE REVIEW

File: TypeCobol.Test/Parser/Programs/Cobol85/WhenTest.rdzMix.txt [hunk 1]
```diff
@@ -25,10 +25,11 @@ Line 21[17,26] <27, Error, Syntax> - Syntax error : Missing statement in "when"
                 when bool1
                 when other continue
            end-evaluate      
+Line 24[12,22] <27, Error, Syntax> - Syntax error : Search statement must have at least one when element.
```
Comment (fm-117) [inline TypeCobol.Test/Parser/Programs/Cobol85/WhenTest.rdzMix.txt:28]: This is not correct here but will be fixed when adressing #2087 and #2088.

File: TypeCobol.Test/Parser/Programs/Cobol85/WhenTestNoEndEvaluate.rdzMix.txt [hunk 2]
```diff
@@ -23,10 +23,11 @@ Line 16[17,20] <27, Error, Syntax> - Syntax error : Missing statement in "when"
 Line 20[17,26] <27, Error, Syntax> - Syntax error : Missing statement in "when" clause
                 when bool1
                 when other continue
+Line 22[12,22] <27, Error, Syntax> - Syntax error : Search statement must have at least one when element.
```
Comment (fm-117) [inline TypeCobol.Test/Parser/Programs/Cobol85/WhenTestNoEndEvaluate.rdzMix.txt:26]: Same remark as above

File: TypeCobol/Compiler/CupParser/TypeCobolProgram.cup [hunk 3]
```diff
@@ -1432,7 +1432,9 @@ rewriteStatement ::= RewriteStatement:stmt
 	;
 
 searchStatementWithBody ::=
-		searchStatement whenSearchConditionClauses
+        searchStatement SearchStatementEnd:end
```
Comment (fm-117) [inline TypeCobol/Compiler/CupParser/TypeCobolProgram.cup:1435]: Indent: could you fix the indentation so as to preserve alignment with next lines ?

Comment (fm-117) [reply TypeCobol/Compiler/CupParser/TypeCobolProgram.cup:1435]: Also another remark, when modifying `TypeCobolProgram.cup`, we should also report the changes into `EmptyTypeCobolProgram.cup` without the parser call `my_parser.Builder.EndSearchStatementWithBody(end);`. `EmptyTypeCobolProgram.cup` is an empty grammar file to help create another parser, it is not currently used but should be kept in-sync.

**Review by fm-117 — CHANGES_REQUESTED** [review]

File: TypeCobol/Compiler/CupParser/TypeCobolProgram.cup (hunk 3 above)
Comment (laabidiHend) [reply TypeCobol/Compiler/CupParser/TypeCobolProgram.cup]: Fixed in ebdc2b7f2d56b9226571d1def1198cd2e6b439af

**Review by mayanje — APPROVED** [review]

**Pull request merged**